library/spdm_requester_lib/libspdm_req_get_key_pair_info.c

```diff
@@ -0,0 +1,266 @@
+/**
+ *  Copyright Notice:
+ *  Copyright 2024 DMTF. All rights reserved.
+ *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
+ **/
+
+#include "internal/libspdm_requester_lib.h"
+
+#if LIBSPDM_ENABLE_CAPABILITY_GET_KEY_PAIR_INFO_CAP
+
+typedef struct {
+    spdm_message_header_t header;
+    uint8_t total_key_pairs;
+    uint8_t key_pair_id;
+    uint16_t capabilities;
+    uint16_t key_usage_capabilities;
+    uint16_t current_key_usage;
+    uint32_t asym_algo_capabilities;
+    uint32_t current_asym_algo;
+    uint16_t public_key_info_len;
+    uint8_t assoc_cert_slot_mask;
+    uint8_t public_key_info[LIBSPDM_MAX_PUBLIC_KEY_INFO_LEN];
+} libspdm_key_pair_info_response_max_t;
+
+/**
+ * This function sends GET_KEY_PAIR_INFO and receives KEY_PAIR_INFO *
+ *
+ * @param  context             A pointer to the SPDM context.
+ *
+ **/
+
+static libspdm_return_t libspdm_try_get_key_pair_info(libspdm_context_t *spdm_context,
+                                                      const uint32_t *session_id,
+                                                      uint8_t key_pair_id
+                                                      )
+{
+    libspdm_return_t status;
+    spdm_get_key_pair_info_request_t *spdm_request;
+    size_t spdm_request_size;
+    libspdm_key_pair_info_response_max_t *spdm_response;
+    size_t spdm_response_size;
+    uint8_t *message;
+    size_t message_size;
+    size_t transport_header_size;
+    libspdm_session_info_t *session_info;
+    libspdm_session_state_t session_state;
+
+    uint16_t key_usage_capabilities;
+    uint16_t current_key_usage;
+    uint32_t asym_algo_capabilities;
+    uint32_t current_asym_algo;
+
+    if (libspdm_get_connection_version(spdm_context) < SPDM_MESSAGE_VERSION_13) {
+        return LIBSPDM_STATUS_UNSUPPORTED_CAP;
+    }
+
+    if ((key_pair_id == 0) || (key_pair_id > SPDM_MAX_SLOT_COUNT)) {
```

Comment:
@steven-bellock, do you agree this check: `key_pair_id > SPDM_MAX_SLOT_COUNT` ?

Reply:
The number of `KeyPairId`s can be larger than the number of certificate slots. But we can run that by the SPDM Working Group to ensure that's what was intended in the specification.